aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -850,6 +850,10 @@ impl AptosVM {
             )])?;
         }
 
+        if entry_fn.module().address.is_special() {
```

Comment:
Can we move this marking to before we charge gas? i.e. before `// Note: Feature gating is needed ... ` otherwise it just breaks into gas charging randomly 😅

Q: what about view functions, are these bulletproofs allowed there?
Q: can you call them in `init_module`? I.e. call an entry function `0x1::code::publish_txn` which publishes code and then calls 0x234::some_module_init_module` using bulletproofs? Is this expected?

Reply:
1. I guess they would not be allowed in non-0x1 view functions. This is fine. Generally, view functions should not need to have to verify ZKPs.

2. Interesting possibility. But then the batch Bulletproof call would come from 0x234, no? So it would be restricted, no?

Reply:
@georgemitenkov, does this cover us from all angles? e.g., multisig_v2 TXNs, <other weird TXN types> / etc?

Reply:
`validate_and_execute_entry_function` will be called from multisig as well, so it does ensure that you can only call these natives from `0x1::..` entries 👍